src/algebraic_geometry/morphisms/quasi_compact.lean

```diff
@@ -91,6 +96,129 @@ begin
+  refine (set.inter_eq_right_iff_subset.mpr (RingedSpace.basic_open_subset _ _)).symm.trans _,
+  rw [e, set.Union₂_inter],
+  apply le_antisymm; apply set.Union₂_subset,
+  { intros i hi,
+    refine set.subset.trans _ (set.subset_Union₂ _ (set.mem_range_self ⟨i, hi⟩)),
+    exact set.subset.rfl },
+  { rintro ⟨i, hi⟩ ⟨⟨j, hj⟩, hj'⟩,
+    rw ← hj',
+    refine set.subset.trans _ (set.subset_Union₂ j hj),
+    exact set.subset.rfl }
+end
+
+lemma quasi_compact_affine_property_is_local :
+  affine_target_morphism_property.is_local quasi_compact.affine_property :=
+begin
+  split,
+  { apply affine_target_morphism_property.respects_iso_mk; rintros X Y Z _ _ _ H,
+    exacts [@@homeomorph.compact_space _ _ H (Top.homeo_of_iso (as_iso e.inv.1.base)), H] },
+  { introv H,
+    delta quasi_compact.affine_property at H ⊢,
+    change compact_space ((opens.map f.val.base).obj (Y.basic_open r)),
+    rw Scheme.preimage_basic_open f r,
+    erw ← is_compact_iff_compact_space,
+    rw ← is_compact_univ_iff at H,
+    exact is_compact_basic_open X H _ },
+  { rintros X Y H f S hS hS',
+    resetI,
+    rw ← is_affine_open.basic_open_union_eq_self_iff at hS,
+    delta quasi_compact.affine_property,
+    rw ← is_compact_univ_iff,
+    change is_compact ((opens.map f.val.base).obj ⊤).1,
+    rw ← hS,
+    dsimp [opens.map],
+    simp only [opens.coe_supr, set.preimage_Union, subtype.val_eq_coe],
+    exacts [compact_Union (λ i, is_compact_iff_compact_space.mpr (hS' i)), top_is_affine_open _] }
+end
+
+lemma quasi_compact.affine_open_cover_tfae {X Y : Scheme.{u}} (f : X ⟶ Y) :
+  tfae [quasi_compact f,
+    ∃ (𝒰 : Scheme.open_cover.{u} Y) [∀ i, is_affine (𝒰.obj i)],
+      ∀ (i : 𝒰.J), compact_space (pullback f (𝒰.map i)).carrier,
+    ∀ (𝒰 : Scheme.open_cover.{u} Y) [∀ i, is_affine (𝒰.obj i)] (i : 𝒰.J),
+      compact_space (pullback f (𝒰.map i)).carrier,
+    ∀ {U : Scheme} (g : U ⟶ Y) [is_affine U] [is_open_immersion g],
+      compact_space (pullback f g).carrier,
+    ∃ {ι : Type u} (U : ι → opens Y.carrier) (hU : supr U = ⊤) (hU' : ∀ i, is_affine_open (U i)),
+      ∀ i, compact_space (f.1.base ⁻¹' (U i).1)] :=
+quasi_compact_eq_affine_property.symm ▸
+  quasi_compact_affine_property_is_local.affine_open_cover_tfae f
+
+lemma quasi_compact.open_cover_tfae {X Y : Scheme.{u}} (f : X ⟶ Y) :
+  tfae [quasi_compact f,
+    ∃ (𝒰 : Scheme.open_cover.{u} Y), ∀ (i : 𝒰.J),
+      quasi_compact (pullback.snd : (𝒰.pullback_cover f).obj i ⟶ 𝒰.obj i),
+    ∀ (𝒰 : Scheme.open_cover.{u} Y) (i : 𝒰.J),
+      quasi_compact (pullback.snd : (𝒰.pullback_cover f).obj i ⟶ 𝒰.obj i),
+    ∀ (U : opens Y.carrier), quasi_compact (f ∣_ U),
+    ∀ {U : Scheme} (g : U ⟶ Y) [is_open_immersion g],
+      quasi_compact (pullback.snd : pullback f g ⟶ _),
+    ∃ {ι : Type u} (U : ι → opens Y.carrier) (hU : supr U = ⊤), ∀ i, quasi_compact (f ∣_ (U i))] :=
+quasi_compact_eq_affine_property.symm ▸
+  quasi_compact_affine_property_is_local.target_affine_locally_is_local.open_cover_tfae f
```

Comment:
Should these two be merged? Probably already at the general level of local properties?

Reply:
I don't think it is necessary. The use case of these lemmas are usually to and from `quasi_compact`, and their proofs are separate as well.